sdk/storage/azure-storage-common/src/xml_wrapper.cpp

```diff
@@ -4,13 +4,381 @@
 #include "azure/storage/common/internal/xml_wrapper.hpp"
 
 #include <limits>
+#include <memory>
 #include <stdexcept>
 
+#include <azure/core/platform.hpp>
+
+#if defined(AZ_PLATFORM_WINDOWS)
+#if !defined(WIN32_LEAN_AND_MEAN)
+#define WIN32_LEAN_AND_MEAN
+#endif
+#if !defined(NOMINMAX)
+#define NOMINMAX
+#endif
+#include <webservices.h>
+#else
 #include <libxml/xmlreader.h>
 #include <libxml/xmlwriter.h>
+#endif
 
 namespace Azure { namespace Storage { namespace _internal {
 
+#if defined(AZ_PLATFORM_WINDOWS)
+
+  struct XmlReaderContext
+  {
+    XmlReaderContext()
+    {
+      HRESULT ret = WsCreateError(nullptr, 0, &error);
+      if (ret != NO_ERROR)
+      {
+        throw std::runtime_error("Failed to initialize xml reader.");
+      }
+      ret = WsCreateReader(nullptr, 0, &reader, error);
+      if (ret != NO_ERROR)
+      {
+        WsFreeError(error);
+        throw std::runtime_error("Failed to initialize xml reader.");
+      }
+    }
+    XmlReaderContext(const XmlReaderContext&) = delete;
+    XmlReaderContext& operator=(const XmlReaderContext&) = delete;
+    ~XmlReaderContext()
+    {
+      WsFreeReader(reader);
+      WsFreeError(error);
+    }
+
+    WS_XML_READER* reader = nullptr;
+    WS_ERROR* error = nullptr;
+    bool readingAttributes = false;
+    ULONG attributeIndex = 0;
+    const WS_XML_ELEMENT_NODE* attributeElementNode = nullptr;
+  };
+
+  XmlReader::XmlReader(const char* data, size_t length)
+  {
+    if (length > static_cast<size_t>(std::numeric_limits<ULONG>::max()))
+    {
+      throw std::runtime_error("Xml data too big.");
+    }
+
+    auto context = std::make_unique<XmlReaderContext>();
+
+    WS_XML_READER_BUFFER_INPUT bufferInput;
+    ZeroMemory(&bufferInput, sizeof(bufferInput));
+    bufferInput.input.inputType = WS_XML_READER_INPUT_TYPE_BUFFER;
+    bufferInput.encodedData = const_cast<char*>(data);
+    bufferInput.encodedDataSize = static_cast<ULONG>(length);
+    WS_XML_READER_TEXT_ENCODING textEncoding;
+    ZeroMemory(&textEncoding, sizeof(textEncoding));
+    textEncoding.encoding.encodingType = WS_XML_READER_ENCODING_TYPE_TEXT;
+    textEncoding.charSet = WS_CHARSET_AUTO;
+    HRESULT ret = WsSetInput(
```

Comment:
How's the performance? This is something @JeffreyRichter had called out earlier:
https://github.com/Azure/azure-sdk-for-cpp/issues/1697#issuecomment-781651511

Reply:
rough measurement shows that it's 3~4 times faster than libxml2.

Reply:
Does it have a noticeable e2e impact on things like blob upload/download?

Reply:
> Does it have a noticeable e2e impact on things like blob upload/download?

I don't think so. Only list operations return xml body, the size of which is usually several MB. Parsing this size of data can be really fast compared to the time spent on network transfer. So there's no noticeable impact on e2e.